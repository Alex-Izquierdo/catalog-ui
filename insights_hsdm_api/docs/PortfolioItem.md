# InsightsServiceCatalogApi.PortfolioItem

## Properties
Name | Type | Description | Notes
------------ | ------------- | ------------- | -------------
**portfolioId** | **Number** |  | [optional] 
**favorite** | **Boolean** |  | [optional] 
**name** | **String** |  | [optional] 
**description** | **String** |  | [optional] 
**orphan** | **Boolean** |  | [optional] 
**state** | **String** |  | [optional] 


