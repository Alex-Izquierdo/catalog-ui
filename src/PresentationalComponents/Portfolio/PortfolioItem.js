import React from 'react';
import { withRouter } from 'react-router-dom';
import { connect } from 'react-redux';
import './portfolioitem.scss';
import propTypes from 'prop-types';
import CatItemSvg from '../../assets/images/vendor-openshift.svg';
import ImageWithDefault from '../ImageWithDefault';
import { PortfolioStore } from "../../Store/Actions/PortfolioActions";
import { hideModal, showModal } from "../../Store/Actions/MainModalActions";
import { GridItem, Card, CardHeader, CardBody, CardFooter } from '@patternfly/react-core';
import {bindMethods} from "../../Helpers/Shared/Helper";


const propLine = (prop, value) => {
    return(<div className = "card_element"> {value} </div>);
};

const toDisplayProperty = property => {
    return ['description'].includes(property)
};

const propDetails = item => {
    let details = [];

    for (let property in item) {
        if (item.hasOwnProperty(property) && toDisplayProperty(property)) {
            if (item[property] && item[property] !== undefined) {
                details.push(propLine(property, item[property].toString()));
            }
        }
    }
    return details;
};

const itemDetails = props => {
    let details = propDetails(props);
    return (
        <React.Fragment>
            <div>{details}</div>
        </React.Fragment>
    );
};

const mapDispatchToProps = dispatch => {
    return {
        hideModal: () => dispatch(hideModal()),
        showModal: (modalProps, modalType) => {
            dispatch(showModal({ modalProps, modalType }))
        }
    };
};


class PortfolioItem extends React.Component {
    constructor(props) {
        super(props);
        this.handleOnClick = this.handleOnClick.bind(this);
    };

    handleOnClick() {
        console.log( 'Before OrderService');
        this.setState({ showOrder: true });
        this.props.showModal({
            open: true,
            servicedata: this.props,
            closeModal: this.props.hideModal
        }, 'order');
    };

    render() {
        return (
            <GridItem GridItem sm={6} md={4} lg={4} xl={3}>
                <Card>
                    <div onClick={ () => {this.handleOnClick(this.props)}}>
                        <CardHeader className="card_header">
                            <ImageWithDefault src={this.props.imageUrl || CatItemSvg} defaultSrc={CatItemSvg} width="50" height="50" />
                        </CardHeader>
                        <CardBody className="card_body">
                            <h4>{this.props.name}</h4>
                            {itemDetails(this.props)}
                        </CardBody>
                        <CardFooter>
                        </CardFooter>
                    </div>
                </Card>
            </GridItem>
        );
    };
}

PortfolioItem.propTypes = {
    history: propTypes.object,
    catalog_id: propTypes.string
};

export default withRouter(
    connect(
        null,
        mapDispatchToProps)(PortfolioItem)
);

