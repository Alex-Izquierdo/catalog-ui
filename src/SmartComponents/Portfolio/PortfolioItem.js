import React, { Fragment } from 'react';
import { Link } from 'react-router-dom';
import PropTypes from 'prop-types';
import CatItemSvg from '../../assets/images/vendor-openshift.svg';
import ImageWithDefault from '../../PresentationalComponents/Shared/ImageWithDefault';
import { GridItem, Card, CardHeader, CardFooter } from '@patternfly/react-core';
import ServiceOfferingCardBody from '../../PresentationalComponents/Shared/service-offering-body';
import CardCheckbox from '../../PresentationalComponents/Shared/CardCheckbox';
import './portfolioitem.scss';

const PortfolioItem = props => {
  const renderCardContent = () => (
    <Fragment>
      <CardHeader className="card_header">
        { props.isSelectable && <CardCheckbox
          handleCheck={ () => props.onSelect(props.id) }
          isChecked={ props.isSelected }
          id={ props.id } />
        }
        <ImageWithDefault src={ props.imageUrl || CatItemSvg } width="30" height="20" />
      </CardHeader>
      <ServiceOfferingCardBody { ...props }/>
      <CardFooter>
      </CardFooter>
    </Fragment>
  );
  return (
    <GridItem sm={ 6 } md={ 4 } lg={ 4 } xl={ 3 }>
      <Card>
        { props.isSelectable ? renderCardContent() : (
          <Link to={ props.orderUrl } className="card-link" >
            { renderCardContent() }
          </Link>
        ) }
      </Card>
    </GridItem>
  );};

PortfolioItem.propTypes = {
  history: PropTypes.object,
  showModal: PropTypes.func,
  hideModal: PropTypes.func,
  imageUrl: PropTypes.string,
  name: PropTypes.string,
  id: PropTypes.string,
  isSelectable: PropTypes.bool,
  isSelected: PropTypes.bool,
  onSelect: PropTypes.func,
  orderUrl: PropTypes.string.isRequired
};

export default PortfolioItem;
