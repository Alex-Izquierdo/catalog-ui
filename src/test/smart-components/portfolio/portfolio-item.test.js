import React from 'react';
import { shallow, mount } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import { MemoryRouter } from 'react-router-dom';

import PortfolioItem from '../../../smart-components/portfolio/portfolio-item';

describe('<PortfolioItem />', () => {
  let initialProps;

  beforeEach(() => {
    initialProps = {
      orderUrl: '/order',
      id: '1',
      name: 'Foo',
      description: 'Bar',
      display_name: 'quux'
    };
  });

  it('should render correctly', () => {
    const wrapper = shallow(<PortfolioItem { ...initialProps } />);
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });

  it('should check the item corecttly', () => {
    const onSelect = jest.fn();
    const wrapper = mount(
      <MemoryRouter>
        <PortfolioItem { ...initialProps } onSelect={ onSelect } isSelectable />
      </MemoryRouter>
    );
    wrapper.find('input').simulate('change');
    expect(onSelect).toHaveBeenCalledWith('1');
  });
});
