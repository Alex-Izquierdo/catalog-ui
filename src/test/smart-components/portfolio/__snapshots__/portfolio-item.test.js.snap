// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PortfolioItem /> should render correctly 1`] = `
<PortfolioItem
  description="Bar"
  display_name="quux"
  id="1"
  name="Foo"
  orderUrl="/order"
>
  <Styled(GalleryItem)>
    <GalleryItem
      className="sc-AxgMl beIutK"
    >
      <div
        className="sc-AxgMl beIutK"
      >
        <Styled(Card)>
          <Card
            className="sc-AxhUy kEBXQD"
          >
            <article
              className="pf-c-card sc-AxhUy kEBXQD"
            >
              <CardHeader
                className="card_header"
              >
                <div
                  className="pf-c-card__header pf-c-title pf-m-md card_header"
                >
                  <Level>
                    <div
                      className="pf-l-level"
                    >
                      <CardIcon
                        height={40}
                        src="/api/catalog/v1.0/portfolio_items/1/icon"
                        style={Object {}}
                      >
                        <styled.div
                          height={40}
                        >
                          <div
                            className="sc-AxirZ kGKGpr"
                            height={40}
                          >
                            <IconPlaceholder
                              height={40}
                            >
                              <svg
                                height={40}
                                width={40}
                              >
                                <circle
                                  cx={20}
                                  cy={20}
                                  fill="#ecebeb"
                                  r={20}
                                />
                              </svg>
                            </IconPlaceholder>
                            <Styled(t)
                              delayTime={0}
                              height={0}
                              onError={[Function]}
                              onLoad={[Function]}
                              src="/api/catalog/v1.0/portfolio_items/1/icon"
                              threshold={2000}
                            >
                              <t
                                afterLoad={[Function]}
                                beforeLoad={[Function]}
                                className="sc-AxjAm kNrCzh"
                                delayMethod="throttle"
                                delayTime={0}
                                effect=""
                                height={0}
                                onError={[Function]}
                                onLoad={[Function]}
                                placeholderSrc=""
                                src="/api/catalog/v1.0/portfolio_items/1/icon"
                                threshold={2000}
                                visibleByDefault={false}
                                wrapperClassName=""
                              >
                                <t
                                  afterLoad={[Function]}
                                  beforeLoad={[Function]}
                                  className="sc-AxjAm kNrCzh"
                                  delayMethod="throttle"
                                  delayTime={0}
                                  height={0}
                                  threshold={2000}
                                  visibleByDefault={false}
                                >
                                  <img
                                    className="sc-AxjAm kNrCzh"
                                    height={0}
                                    onError={[Function]}
                                    onLoad={[Function]}
                                    src="/api/catalog/v1.0/portfolio_items/1/icon"
                                  />
                                </t>
                              </t>
                            </Styled(t)>
                          </div>
                        </styled.div>
                      </CardIcon>
                    </div>
                  </Level>
                </div>
              </CardHeader>
              <ServiceOfferingCardBody
                description="Bar"
                display_name="quux"
                id="1"
                name="Foo"
                orderUrl="/order"
              >
                <CardBody
                  className="card-height"
                >
                  <div
                    className="pf-c-card__body card-height"
                  >
                    <TextContent>
                      <div
                        className="pf-c-content"
                      >
                        <ConditionalLink>
                          <Text
                            component="h3"
                            title="Foo"
                          >
                            <h3
                              className=""
                              data-pf-content={true}
                              title="Foo"
                            >
                              <styled.div>
                                <div
                                  className="sc-AxhCb hYKqDj"
                                >
                                  Foo
                                </div>
                              </styled.div>
                            </h3>
                          </Text>
                        </ConditionalLink>
                        <Text
                          component="small"
                        >
                          <small
                            className=""
                            data-pf-content={true}
                          >
                            Â 
                          </small>
                        </Text>
                      </div>
                    </TextContent>
                    <ItemDetails
                      description="Bar"
                      id="1"
                      orderUrl="/order"
                      toDisplay={
                        Array [
                          "description",
                        ]
                      }
                    >
                      <styled.div
                        key="card-prop-description"
                      >
                        <div
                          className="sc-AxiKw iUXPLF"
                        >
                          Bar
                        </div>
                      </styled.div>
                    </ItemDetails>
                  </div>
                </CardBody>
              </ServiceOfferingCardBody>
              <CardFooter>
                <div
                  className="pf-c-card__footer"
                />
              </CardFooter>
            </article>
          </Card>
        </Styled(Card)>
      </div>
    </GalleryItem>
  </Styled(GalleryItem)>
</PortfolioItem>
`;
